src/Routing/Router.php

```diff
@@ -650,18 +650,29 @@ public static function reverseToArray($params): array
         }
         $pass = $params['pass'] ?? [];
 
+        $template = $params['_matchedRoute'] ?? null;
```

Comment:
In the past I have wondered if storing the full route instance instead of just the template would be useful. It would have been for this case :). Perhaps something to consider for 5.0.

Reply:
Having the full route in the request (perhaps as an attribute) would be a nice improvement.